File: app/src/main/java/com/kakeibo/util/UtilSubscription.java
package com.kakeibo.util;

import android.content.res.Resources;

import com.kakeibo.data.SubscriptionStatus;
import com.kakeibo.R;
import com.kakeibo.billing.BillingUtilities;

public class UtilSubscription {

    /**
     * Return the resource string for the basic subscription button.
     *
     * Add an asterisk if the subscription is not local and might not be modifiable on this device.
     */
    public static String basicTextForSubscription(Resources res, SubscriptionStatus subscription) {
        String text;

        if (BillingUtilities.isAccountHold(subscription)) {
            text = res.getString(R.string.subscription_option_basic_message_account_hold);
        } else if (BillingUtilities.isGracePeriod(subscription)) {
            text = res.getString(R.string.subscription_option_basic_message_grace_period);
        } else if (BillingUtilities.isSubscriptionRestore(subscription)) {
            text = res.getString(R.string.subscription_option_basic_message_restore);
        } else if (BillingUtilities.isBasicContent(subscription)) {
            text = res.getString(R.string.subscription_option_basic_message_current);
        } else {
            text =  res.getString(R.string.subscription_option_basic_message);
        }
        if (subscription.isLocalPurchase) {
            return text;
        } else {
            // No local record, so the subscription cannot be managed on this device.
            return text + "*";
        }
    }

    /**
     * Return the resource string for the premium subscription button.
     *
     * Add an asterisk if the subscription is not local and might not be modifiable on this device.
     */
    public static String premiumTextForSubscription(Resources res,
                                                    SubscriptionStatus subscription) {
        String text;
        if (BillingUtilities.isAccountHold(subscription)) {
            text = res.getString(R.string.subscription_option_premium_message_account_hold);
        } else if (BillingUtilities.isGracePeriod(subscription)) {
            text = res.getString(R.string.subscription_option_premium_message_grace_period);
        } else if (BillingUtilities.isSubscriptionRestore(subscription)) {
            text =res.getString(R.string.subscription_option_premium_message_restore);
        } else if (BillingUtilities.isPremiumContent(subscription)) {
            text = res.getString(R.string.subscription_option_premium_message_current);
        } else {
            text = res.getString(R.string.subscription_option_premium_message);
        }

        if (subscription.isLocalPurchase) {
            return text;
        } else {
            // No local record, so the subscription cannot be managed on this device.
            return text + "*";
        }
    }
}

File: app/src/main/java/com/kakeibo/util/UtilSubscription.kt
package com.kakeibo.util

import android.content.res.Resources
import com.kakeibo.R
import com.kakeibo.billing.BillingUtilities.isAccountHold
import com.kakeibo.billing.BillingUtilities.isBasicContent
import com.kakeibo.billing.BillingUtilities.isGracePeriod
import com.kakeibo.billing.BillingUtilities.isPremiumContent
import com.kakeibo.billing.BillingUtilities.isSubscriptionRestore
import com.kakeibo.data.Subscription

object UtilSubscription {
    /**
     * Return the resource string for the basic subscription button.
     *
     * Add an asterisk if the subscription is not local and might not be modifiable on this device.
     */
    fun basicTextForSubscription(res: Resources, subscription: Subscription): String {
        val text: String = if (isAccountHold(subscription)) {
            res.getString(R.string.subscription_option_basic_message_account_hold)
        } else if (isGracePeriod(subscription)) {
            res.getString(R.string.subscription_option_basic_message_grace_period)
        } else if (isSubscriptionRestore(subscription)) {
            res.getString(R.string.subscription_option_basic_message_restore)
        } else if (isBasicContent(subscription)) {
            res.getString(R.string.subscription_option_basic_message_current)
        } else {
            res.getString(R.string.subscription_option_basic_message)
        }
        return if (subscription.isLocalPurchase) {
            text
        } else {
            // No local record, so the subscription cannot be managed on this device.
            "$text*"
        }
    }

    /**
     * Return the resource string for the premium subscription button.
     *
     * Add an asterisk if the subscription is not local and might not be modifiable on this device.
     */
    fun premiumTextForSubscription(res: Resources,
                                   subscription: Subscription): String {
        val text: String = if (isAccountHold(subscription)) {
            res.getString(R.string.subscription_option_premium_message_account_hold)
        } else if (isGracePeriod(subscription)) {
            res.getString(R.string.subscription_option_premium_message_grace_period)
        } else if (isSubscriptionRestore(subscription)) {
            res.getString(R.string.subscription_option_premium_message_restore)
        } else if (isPremiumContent(subscription)) {
            res.getString(R.string.subscription_option_premium_message_current)
        } else {
            res.getString(R.string.subscription_option_premium_message)
        }
        return if (subscription.isLocalPurchase) {
            text
        } else {
            // No local record, so the subscription cannot be managed on this device.
            "$text*"
        }
    }
}